<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Transitional//EN" "http://www.w3.org/TR/xhtml1/DTD/xhtml1-transitional.dtd">
<html xmlns="http://www.w3.org/1999/xhtml">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<link rel="stylesheet" type="text/css" href="style1.css" />
<title>RNA-Seq Analysis of BES on <i>Rattus norvegicus</i> liver genes</title>
</head>

<body>
<div id="container">
		<div id="header">
            <h1>Finge<span class="off">RNA</span>ils</h>
            <h2>Database of BES action on the expression of <i>Rattus norvegicus</i> liver genes</h2>
        </div>   
        
        <div id="menu">
        	<ul>
            	<li class="menuitem"><a href="index.html">Home</a></li>
                <li class="menuitem"><a href="about.html">About</a></li>
                <li class="menuitem"><a href="http://143.210.153.168:3838/rn6_app/">Analysis and Visualisation</a></li>
 		<li class="menuitem"><a href="mapping.html">Annotation Mapping</a></li>
            </ul>
        </div>
        
        <div id="leftmenu">

        <div id="leftmenu_top"></div>

				<div id="leftmenu_main">    
                
                <h3>Links</h3>
                        
                <ul>
                    <li><a href="http://www.ncbi.nlm.nih.gov/pubmed/25150839">Wang Paper</a></li>
                    <li><a href="http://cole-trapnell-lab.github.io/cufflinks/">Cufflinks</a></li>
                    <li><a href="https://ccb.jhu.edu/software/tophat/index.shtml">TopHat</a></li>
                    <li><a href="https://github.com/RoryBioinformatics/Steered_Project">GitHub Resources</a></li>
                </ul>
</div>
                
                
              <div id="leftmenu_bottom"></div>
        </div>
        
        
        
        
		<div id="content">
        
        
        <div id="content_top"></div>
        <div id="content_main">
<h2>Summary of BES RNA-Seq analysis methods</h2>
        	<p>&nbsp;</p>
           	<p>&nbsp;</p>
<h3>The Wang et al. Paper (2014)</h3>
		<p>The article upon which this website and its' analysis is based upon is the 2014 article by Wang <i>et al.</i>, <i>"A comprehensive study design reveals treatment- and transcript abundanceâ€“dependent concordance between RNA-seq and microarray data".</i> The article initially aims to compare the effects of a set of chemicals with 5 different modes of action (MOAs) on data generated by both RNA-seq and Microarray high throughput technologies. To test the two technologies, Sprague-Dawley Rats were exposed to 1 of 27 chemicals with corresponding controls and their RNA was extracted from liver cells. 15 of these chemicals were used as a training set and 12 as a test set for the creation of predicative expression models from the data gained in order to test both pipelines and the comparative technologies for their efficaciousness. Our analysis used both the original Wang paper's genome (Rn4) and the latest build (Rn6) of the rat genome to give an updated analysis.</p>
		<p>&nbsp;</p>
<h3>Pipeline Choice</h3>
        	<p>The Wang et al (2014) article used 6 different pipelines in the analysis of the RNA-Seq and Microarray data. Of these, the P5 analysis pipeline was chosen for our RNA-Seq analysis. It included aspects of the tuxedo package, TopHat, Bowtie and Cufflinks with the additional CummeRbund R package. This commonly used pipeline was preferred due to a reported 2nd highest level of genes found and highest Microarray-RNA-Seq concurrence within the article.</p>
        	<p>&nbsp;</p>
<h3>Chemical Choice</h3>
        	<p> The FingeRNAils database was designed with the aim of exploring the chemical effects of Beta-Estradiol (BES) on <i>Rattus norvegicus</i> liver cells. Within the Wang (2014) experimental procedure, the chemical was used as part of the test group of chemicals at a concentration of 150mg/kg/day for five days. BES's MOA is that of an estrogen receptor agonist. The choice of this chemical was made because it had the highest quality RNA in the runs made and therefore a high proportion of reads with a high quality Phred score. Matching controls were chosen too, based on the use of BES.</p>
        	<p>&nbsp;</p>
<h3>Results of Analysis</h3>
		<p>The Wang paper states that an average TopHat mapping rate per sample of 0.808 (0.875 max., 0.722 min.)Our group found an average of 0.907 (0.937 max., 0.892 min.) when mapping to the rn4 genome and 0.951 (0.978 max., 0.934 min.) when mapping to the rn6 genome. The authors then run Cufflinks without a reference annotation to assemble transcripts from the mapped reads identifying on average 19,309 transcripts per sample. Using Cufflinks, our group identified an average of 32,594 and 32,988 transcripts per sample for the rn4 and rn6 mapped datasets respectively. </p><p>&nbsp;</p>
<table style="width:100%">
  <tr>
    <th>Global Features</th>
    <th>Rn4</th>
    <th>Rn6</th>
  </tr>
  <tr>
    <td>Genes</td>
    <td>18794</td>
    <td>19317</td>
  </tr>
  <tr>
    <td>Promoters</td>
    <td>18794</td>
    <td>19317</td>
  </tr>
<tr>
    <td>TSS</td>
    <td>20492</td>
    <td>21136</td>
  </tr>
<tr>
    <td>Total Isoforms</td>
    <td>23903</td>
    <td>24718</td>
  </tr>
<tr>
    <td>Putative Novel Isoforms</td>
    <td>1898</td>
    <td>1876</td>
  </tr>
</table><p>&nbsp;</p>
<p>Global Feature Table: This compares the overall results obtained by pipeline 5. The only difference between the two datasets is the use of either the rn4 or rn6 genomes. Note the number of genes reflects the number of promoters. There are 1698 and 1819 more transcription start sites (TSS) than promoters respectively suggesting two levels of regulation: pre and post transcriptional. Novel isoforms here are defined as isoforms identified that have no associated gene name in the UCSC annotated genome however these have not been validated.</p><p>&nbsp;</p><p>&nbsp;</p>

<image src="images1/peaks.png"><p>&nbsp;</p><p>Figure 2: Global gene and isoform density. Distribution of FPKM values for each condition in the rn4 (A) and rn6 (B) datasets</p><p>&nbsp;</p><p>&nbsp;</p>
<image src="images1/line.png"><p>&nbsp;</p><p>Figure 3: Global differential expression of genes. Comparison of FPKM values for each gene between conditions at both the gene and isoform level in the rn4 (A) and rn6 (B) datasets. Each black dot represents a single gene. A black dot on the diagonal dotted line indicates no differential expression of the gene. The blue line is the line of best fit that best represents the overall relationship between the two conditions. When the blue line falls below the dotted line as FPKM increases, this indicates that the greatest overall effect of treatment is to reduce the expression of many highly expressed genes/isoforms. This overall effect in gene expression is eliminated completely by the update in the rat genome from rn4 to rn6 while the overall effect in isoform expression is reduced but not eliminated. </p><p>&nbsp;</p><p>&nbsp;</p>
<image src="images1/volcano.png"><p>&nbsp;</p><p>Figure 4: Significant differentially expressed of genes. The volcano plots show the differential expression of each gene/isoform as a function of significance. Genes with a q-value (corrected p-value) of less than 0.05 are significant and are shown as red dots. The genes/isoforms in the upper right and left corners of each volcano plot are those that are both highly significant and have the greatest change in expression. The significant differentially expressed subset of the rn4 dataset contains 301 known (plus 45 putative novel) genes and 168 known (plus 30 putative novel) isoforms. The significant differentially expressed subset of the rn6 dataset contains 409 known (plus 60 putative novel) genes and 271 known (plus 35 putative novel) isoforms.</p><p>&nbsp;</p>


</div>
        <div id="content_bottom"></div>
            
            <div id="footer"><h3><a href="http://www.bryantsmith.com">florida web design</a></h3></div>
      </div>
   </div>
</body>
</html>
